src/ngx_http_lua_balancer.c

```diff
@@ -409,7 +409,7 @@ ngx_http_lua_balancer_free_peer(ngx_peer_connection_t *pc, void *data,
 
 int
 ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
-    const u_char *addr, size_t addr_len, int port, char **err)
+    const u_char *addr, size_t addr_len, in_port_t port, char **err)
```

Comment:
We don't want to use `in_port_t` as the parameter type here since this C function is part of the ABI where we want to stick with very basic C data types.  Let's do proper type coercions in the C function body instead.